InnerEye/ML/utils/io_util.py

```diff
@@ -412,25 +412,44 @@ def load_image_in_known_formats(file: Path,
         raise ValueError(f"Unsupported image file type for path {file}")
 
 
-def load_labels_from_dataset_source(dataset_source: PatientDatasetSource) -> np.ndarray:
+def load_labels_from_dataset_source(dataset_source: PatientDatasetSource, check_exclusive: bool = True) -> np.ndarray:
     """
     Load labels containing segmentation binary labels in one-hot-encoding.
     In the future, this function will be used to load global class and non-imaging information as well.
 
     :param dataset_source: The dataset source for which channels are to be loaded into memory.
-    :return A label sample object containing ground-truth information.
+    :param check_exclusive: Check that the labels are mutually exclusive (defaults to True)
+    :return: A label sample object containing ground-truth information.
     """
     labels = np.stack(
         [load_image(gt, ImageDataType.SEGMENTATION.value).image for gt in dataset_source.ground_truth_channels])
 
+    if check_exclusive and not labels_are_mutually_exclusive(labels):
+        raise ValueError(f'The labels for patient {dataset_source.metadata.patient_id} are not mutually exclusive. '
+                         'Some loss functions (e.g. SoftDice) may produce results on overlapping labels, while others (e.g. FocalLoss) will fail. '
+                         'If you are sure that you want to use mutually exclusive labels, '
+                         'then re-run with the check_exclusive flag set to false in the settings file. '
+                         'Note that this is the first error encountered, other samples/patients may also have overlapping labels.')
+
     # Add the background binary map
     background = np.ones_like(labels[0])
     for c in range(len(labels)):
         background[labels[c] == 1] = 0
-    background = background[None, ...]
+    background = background[np.newaxis, ...]
     return np.vstack((background, labels))
 
 
+def labels_are_mutually_exclusive(labels: List[np.ndarray]) -> bool:
+    '''
+    Many loss functions need the segmentations to be mutually exclusive (e.g. FocalLoss) while others are more forgiving
+    This function checks the labels to ensure that they are mutually exclusive.
+
+    :param labels: The list of labels
+    :returns: True if the labels are mutually exclusive
```

Comment:
Please describe the inputs in more details. What are the expected tensor sizes? Also say that each entry in the list corresponds to one class  (I think it is, right?)

Reply:
Since this is only called from within `load_labels_from_dataset_source` and is just a single line of code I have moved the code into the calling function and deleted this function.